tensorflow_addons/optimizers/rectified_adam.py

```diff
@@ -123,7 +127,16 @@ def __init__(
         self._set_hyper("decay", self._initial_decay)
         self._set_hyper("weight_decay", weight_decay)
         self._set_hyper("sma_threshold", sma_threshold)
-        self._set_hyper("total_steps", float(total_steps))
+        if isinstance(total_steps, float):
+            warnings.warn(
+                "The parameter `total_steps` passed to the __init__ of RectifiedAdam "
+                "is a float. This behavior is deprecated and in Addons 0.11, this "
```

Comment:
Any reason to select 0.11? Also, how do we keep track of this when we get to version 0.11?

Reply:
We need to give a date in the warning, otherwise users won't care. They'll just say: "I'll live tomorrow's probems to tomorrow's me". :P

Reply:
Milestones require a due date, not version. 

Why not provide a deprecation date instead of a version? But this directly means we'll need to create a new release on that due date.

Reply:
When stopping support for a branch, it's indeed common to specify a branch: ex the python org will stop support for the branch 2.7 of python january 1st 2020. 

When deprecating a feature and removing it later, it's more common to use a specific version:
`  /opt/conda/lib/python3.7/site-packages/tensorflow/python/training/tracking/data_structures.py:718: DeprecationWarning:
 Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in
3.9 it will stop working`. 

I believe it makes more sense to use a specific version. We can make a milestone. We're not forced to add a date. For example:
https://github.com/pytest-dev/pytest/milestones

Reply:
Sounds great! I'll accept this PR. We should create a milestone for this as well.

Reply:
Sounds great! I'll accept this PR. We should create a milestone for this as well.

Reply:
Sure, I can do that.

Reply:
Haha, accidentally created a duplicate issue (#1381, closed now).